Test failed on linux/386

I'm using Vagrant.

System: Linux squeeze-i386 2.6.32-5-686 #1 SMP Tue May 13 16:33:32 UTC 2014 i686 GNU/Linux

Error log:

```
godep go test -cover ./...
ok  	github.com/pingcap/tidb	0.105s	coverage: 71.1% of statements
ok  	github.com/pingcap/tidb/column	0.002s	coverage: 82.7% of statements
?   	github.com/pingcap/tidb/context	[no test files]
ok  	github.com/pingcap/tidb/ddl	0.029s	coverage: 82.7% of statements
ok  	github.com/pingcap/tidb/domain	0.005s	coverage: 86.7% of statements
ok  	github.com/pingcap/tidb/expression	0.006s	coverage: 91.9% of statements
ok  	github.com/pingcap/tidb/expression/builtin	0.003s	coverage: 95.7% of statements
ok  	github.com/pingcap/tidb/expression/subquery	0.002s	coverage: 0.0% of statements
ok  	github.com/pingcap/tidb/field	0.002s	coverage: 100.0% of statements
ok  	github.com/pingcap/tidb/infoschema	0.005s	coverage: 98.6% of statements
?   	github.com/pingcap/tidb/interpreter	[no test files]
?   	github.com/pingcap/tidb/kv	[no test files]
ok  	github.com/pingcap/tidb/kv/memkv	1.443s	coverage: 81.8% of statements
ok  	github.com/pingcap/tidb/meta	0.005s	coverage: 85.0% of statements
ok  	github.com/pingcap/tidb/meta/autoid	0.003s	coverage: 89.5% of statements
ok  	github.com/pingcap/tidb/model	0.002s	coverage: 100.0% of statements
ok  	github.com/pingcap/tidb/mysqldef	0.009s	coverage: 89.6% of statements
ok  	github.com/pingcap/tidb/parser	0.008s	coverage: 27.1% of statements
?   	github.com/pingcap/tidb/parser/coldef	[no test files]
ok  	github.com/pingcap/tidb/parser/opcode	0.001s	coverage: 100.0% of statements
?   	github.com/pingcap/tidb/plan	[no test files]
ok  	github.com/pingcap/tidb/plan/plans	0.043s	coverage: 76.5% of statements
?   	github.com/pingcap/tidb/rset	[no test files]
ok  	github.com/pingcap/tidb/rset/rsets	0.011s	coverage: 87.5% of statements
ok  	github.com/pingcap/tidb/sessionctx	0.002s	coverage: 100.0% of statements
ok  	github.com/pingcap/tidb/sessionctx/db	0.001s	coverage: 100.0% of statements
?   	github.com/pingcap/tidb/sessionctx/forupdate	[no test files]
ok  	github.com/pingcap/tidb/sessionctx/variable	0.002s	coverage: 37.1% of statements
?   	github.com/pingcap/tidb/stmt	[no test files]
ok  	github.com/pingcap/tidb/stmt/stmts	0.139s	coverage: 83.7% of statements
ok  	github.com/pingcap/tidb/store/localstore	0.038s	coverage: 80.2% of statements
ok  	github.com/pingcap/tidb/store/localstore/boltdb	0.014s	coverage: 86.4% of statements
?   	github.com/pingcap/tidb/store/localstore/engine	[no test files]
ok  	github.com/pingcap/tidb/store/localstore/goleveldb	0.002s	coverage: 86.7% of statements
ok  	github.com/pingcap/tidb/table	0.009s	coverage: 100.0% of statements
ok  	github.com/pingcap/tidb/table/tables	0.025s	coverage: 75.2% of statements
?   	github.com/pingcap/tidb/tidb-server	[no test files]
ok  	github.com/pingcap/tidb/tidb-server/server	0.132s	coverage: 64.0% of statements
ok  	github.com/pingcap/tidb/util	0.004s	coverage: 73.1% of statements
ok  	github.com/pingcap/tidb/util/arena	0.001s	coverage: 100.0% of statements
ok  	github.com/pingcap/tidb/util/charset	0.001s	coverage: 96.6% of statements
ok  	github.com/pingcap/tidb/util/codec	0.002s	coverage: 89.7% of statements
?   	github.com/pingcap/tidb/util/errors	[no test files]
ok  	github.com/pingcap/tidb/util/errors2	0.001s	coverage: 100.0% of statements
ok  	github.com/pingcap/tidb/util/format	0.001s	coverage: 81.2% of statements
ok  	github.com/pingcap/tidb/util/hack	0.001s	coverage: 92.3% of statements
ok  	github.com/pingcap/tidb/util/mock	0.001s	coverage: 100.0% of statements
?   	github.com/pingcap/tidb/util/mock/mocks	[no test files]
ok  	github.com/pingcap/tidb/util/printer	0.001s	coverage: 89.8% of statements
2015/09/26 18:27:04 etc.go:370: [error] map[int]string

----------------------------------------------------------------------
FAIL: etc_test.go:244: testTypeEtcSuite.TestRoundFloat

etc_test.go:260:
    c.Assert(f, Equals, t.Expect)
... obtained float64 = -1
... expected float64 = -2

2015/09/26 18:27:04 etc.go:192: [error] unkown type 221, binary true
OOPS: 23 passed, 1 FAILED
--- FAIL: TestT (0.00s)
FAIL
coverage: 84.0% of statements
FAIL	github.com/pingcap/tidb/util/types	0.006s
godep: go exit status 1
make: *** [gotest] Error 1
```

--- a/util/types/helper.go
+++ b/util/types/helper.go
@@ -21,19 +21,11 @@ import (
 // RoundFloat uses default rounding mode, see http://www.gnu.org/software/libc/manual/html_node/Rounding.html
 // so we will choose the even number if the result is midway between two representable value.
 // e.g, 1.5 -> 2, 2.5 -> 2.
-func RoundFloat(val float64) float64 {
-	v, frac := math.Modf(val)
-	if val >= 0.0 {
-		if frac > 0.5 || (frac == 0.5 && uint64(v)%2 != 0) {
-			v += 1.0
-		}
-	} else {
-		if frac < -0.5 || (frac == -0.5 && uint64(math.Abs(v))%2 != 0) {
-			v -= 1.0
-		}
+func RoundFloat(f float64) float64 {
+	if math.Remainder(f, 1.0) < 0 {
+		return math.Ceil(f)
 	}
-
-	return v
+	return math.Floor(f)
 }
 
 func getMaxFloat(flen int, decimal int) float64 {

--- a/util/types/helper.go
+++ b/util/types/helper.go
@@ -28,7 +28,7 @@ func RoundFloat(val float64) float64 {
 			v += 1.0
 		}
 	} else {
-		if frac < -0.5 || (frac == -0.5 && uint64(v)%2 != 0) {
+		if frac < -0.5 || (frac == -0.5 && uint64(math.Abs(v))%2 != 0) {
 			v -= 1.0
 		}
 	}

